dockerfiles/hpcc/templates/thor.yaml

```diff
@@ -1,84 +1,137 @@
+      - name: {{ $eclagentName | quote }}
         ports:
           - containerPort: {{ $masterPort }}
         args: [
                 {{ include "hpcc.configArg" . }},
-                {{ include "hpcc.daliArg" $ }},
-                --masterport={{ $masterPort }},
-                --numSlaves={{ $thor.numSlaves }}
+                {{ include "hpcc.daliArg" $ }}
+                {{- if $thor.maxActive }},
+                --maxActive={{ $thor.maxActive }}
+                {{- end }}
               ]
 {{ include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 8 }}
-{{ include "hpcc.addImageAttrs" (dict "root" $ "me" . "imagename" "thormaster") | indent 8 }}
+{{ include "hpcc.addImageAttrs" (dict "root" $ "me" . "imagename" "eclagent") | indent 8 }}
 {{ include "hpcc.addVolumeMounts" . | indent 8 }}
       volumes:
 {{ include "hpcc.addConfigVolume" . | indent 6 }}
 {{ include "hpcc.addVolumes" $ | indent 6 }}
 ---
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: {{ $slaveName | quote }}
-spec:
-  replicas: {{ $thor.numSlaves }}
-  selector:
-    matchLabels:
-      run: {{ $slaveName | quote }}
-  template:
+{{- include "hpcc.generateConfigMap" (dict "root" $ "component" "thor" "me" .) }}
+{{- if $thor.eclagent }}
+    eclagent:
+{{ toYaml $thor.eclagent | indent 6 }}
+      name: myeclagent-thor
+{{- end }}
+  eclagent.yaml: |
+    apiVersion: batch/v1
+    kind: Job
     metadata:
-      labels:
-        run: {{ $slaveName | quote }}
+      name: %jobname
     spec:
-      {{- include "hpcc.checkDataStorageHostMount" (dict "root" $) | indent 6 }}
-      containers:
-        - name: {{ $slaveName | quote }}
-          args: [
-                  {{ include "hpcc.configArg" . }},
-                  {{ include "hpcc.daliArg" $ }},
-                  --slaveport={{ $slavePort }},
-                  --master={{ printf "%s:%v" $serviceName $masterPort }}
-                ]
-{{ include "hpcc.addSecurityContext" (dict "root" $ "me" .) | indent 10 }}
-{{ include "hpcc.addImageAttrs" (dict "root" $ "me" . "imagename" "thorslave") | indent 10 }}
-          ports:
-            - containerPort: {{ $slavePort }}
-{{ include "hpcc.addVolumeMounts" . | indent 10 }}
-      volumes:
-{{ include "hpcc.addConfigVolume" . | indent 6 }}
-{{ include "hpcc.addVolumes" $ | indent 6 }}
----
-apiVersion: v1
-kind: Service
-metadata:
-  name: {{ $serviceName | quote }}
-spec:
-  ports:
-  - port: {{ $masterPort }}
-    protocol: TCP
-    targetPort: {{ $masterPort }}
-  selector:
-    run: {{ .name | quote }}
-  type: ClusterIP
+      ttlSecondsAfterFinished: 100
+      template:
+        spec:
+          serviceAccountName: hpcc
+          {{- include "hpcc.checkDataStorageHostMount" (dict "root" $) | indent 10 }}
+          containers:
+          - name: %jobname
+            image: {{ include "hpcc.imageName" (dict "root" $ "me" . "imagename" "eclagent") }}
+            resources:
+              requests:
+                #request-memory
```

Comment:
was there supposed to be something here?

Reply:
hm,  it's the same as eclagent.yaml - should it not be?

Reply:
I think I had just forgotten how it works...

It gets substituted in from wu values - which you may not want for the eclagent associated with a thor (the wu values probably shoudl go into the thor itself instead)